### PR TITLE
Consistent lowercase dataset names

### LINKED ISSUE #263: Convert all dataset names to lowercase
Keeping with the standards established in other repos - including `tern` and `teal.modules.clinical` - we should convert all dataset names to lowercase (currently all uppercase) in both code and examples. This should not affect the data values (no regeneration needed).

### PR DESCRIPTION
**Pull request by edelarua**
Closes #263 #264

Potential breaking changes: 
Changed parameters `ADSL` to `adsl`, `ADPC` to `adpc`, `event.descr` to `event_descr`, and `censor.descr` to `censor_descr` - I was unable to find any instances where these parameters were explicitly named and the changes would cause an issue. Are there any packages that call the rdata functions instead of using `scda` data?

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -41,6 +41,9 @@
 * Improved usability of examples.
 * Renamed `random.cdisc.data.R` file to `package.R` and cleaned up contents.
 * Added more detailed documentation for function parameters.
+* Converted all dataset names to lowercase/snake case for consistency with other NEST packages.
+* Converted `ADSL` argument to `adsl`, `ADPC` to `adpc`, `event.descr` to `event_descr`,
```
Comment (shajoezhu) [inline NEWS.md:45]: hi @edelarua , can you please list out all the changes of upper cases to lower cases, this is potentially a big break change for users who actually use rcd package, i would like to make this very visible in the NEWS. Thanks!

**Review by shajoezhu — CHANGES_REQUESTED** [review]
@edelarua , can you update the news please! Thanks!

File: NEWS.md (hunk 1 above)
Comment (edelarua) [reply NEWS.md]: I've made a "Breaking changes" section at the top of NEWS to highlight the changes and expanded on which functions the changes apply to.

File: NEWS.md [hunk 2]
```diff
@@ -41,6 +47,7 @@
 * Improved usability of examples.
 * Renamed `random.cdisc.data.R` file to `package.R` and cleaned up contents.
 * Added more detailed documentation for function parameters.
+* Converted all dataset names to lowercase/snake case for consistency with other NEST packages.
```
Comment (Melkiades) [inline NEWS.md:50]: 🎉

**Review by Melkiades — APPROVED** [review]
Lgtm! I checked also on my local and linter is happy. Thanks so much Emily. With all the mods now rcd is much more mature and readable.

**Pull request merged**